### PR TITLE
Add XCTS executable

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Add the elliptic executable to solve the Einstein constraint equations in the XCTS formulation. It supports superposed Kerr-Schild initial data once the apparent-horizon boundary-conditions #2961 are merged. Solves are ridiculously slow until the Multigrid-Schwarz preconditioner is added (currently at #3127).

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

File: src/Elliptic/Actions/InitializeBackgroundFields.hpp [hunk 1]
```diff
@@ -68,8 +68,13 @@ struct InitializeBackgroundFields {
     const auto& background = db::get<BackgroundTag>(box);
```
Comment (nilsdeppe) [inline src/Elliptic/Actions/InitializeBackgroundFields.hpp:70]: could you elaborate in the commit message (briefly) how this fixes anything? I have no idea what to make of this change

**Review by nilsdeppe — CHANGES_REQUESTED** [review]

File: src/Elliptic/Actions/InitializeBackgroundFields.hpp (hunk 1 above)
Comment (nilsvu) [reply src/Elliptic/Actions/InitializeBackgroundFields.hpp:70]: Some background fields need numeric derivatives. The `Binary` analytic data class and the others already support this, but they need the mesh and the inverse Jacobian. The change in this commit makes the action pass those along. I added some comments to the docs and to the commit message.

**Review by nilsdeppe — DISMISSED** [review]

File: src/Elliptic/Executables/Xcts/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  MathFunctions
+  Options
+  Parallel
+  ParallelLinearSolver
+  ParallelNonlinearSolver
+  Xcts
+  XctsAnalyticData
+  XctsBoundaryConditions
+  XctsSolutions
+  Utilities
```
Comment (kidder) [inline src/Elliptic/Executables/Xcts/CMakeLists.txt]: alphabet

File: src/Elliptic/Executables/Xcts/SolveXcts.hpp [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+  static std::string name() noexcept { return "Gmres"; }
+  static constexpr Options::String help = "Options for the GMRES linear solver";
+  using group = LinearSolverGroup;
+};
+}  // namespace SolveXcts::OptionTags
+
+/// \cond
+struct Metavariables {
+  static constexpr size_t volume_dim = 3;
+  static constexpr int conformal_matter_scale = 0;
```
Comment (kidder) [inline src/Elliptic/Executables/Xcts/SolveXcts.hpp:85]: I could not find the documentation for `Xcts::FirstOrderSystem` in order to look up what this int is...

File: src/Elliptic/Executables/Xcts/SolveXcts.hpp [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+  using initial_guess_tag = elliptic::Tags::InitialGuess<
+      ::AnalyticData<volume_dim, initial_guess_registrars>>;
+
+  static constexpr Options::String help{
+      "Find the solution to an XCTS problem."};
+
+  // These are the fields we solve for
+  using fields_tag = ::Tags::Variables<typename system::primal_fields>;
+  // These are the fluxes corresponding to the fields, i.e. essentially their
+  // first derivatives. These are background fields for the linearized sources.
```
Comment (kidder) [inline src/Elliptic/Executables/Xcts/SolveXcts.hpp:115]: just checking: are these both the fluxes and the background fields for the linearized sources?

**Review by kidder — CHANGES_REQUESTED** [review]

File: src/Elliptic/Executables/Xcts/SolveXcts.hpp (hunk 3 above)
Comment (nilsvu) [reply src/Elliptic/Executables/Xcts/SolveXcts.hpp:85]: It's here https://spectre-code.org/structXcts_1_1FirstOrderSystem.html

File: src/Elliptic/Executables/Xcts/SolveXcts.hpp (hunk 4 above)
Comment (nilsvu) [reply src/Elliptic/Executables/Xcts/SolveXcts.hpp:115]: yes. These are the fluxes for the nonlinear DG operator. They act as background fields for the linearized DG operator.

**Comment by nilsvu** [comment]
Squashed in the alphabetization in the CMakeLists.txt and rebased. Thanks for reviewing @nilsdeppe and @kidder !

File: src/Elliptic/Executables/Xcts/SolveXcts.hpp (hunk 3 above)
Comment (kidder) [reply src/Elliptic/Executables/Xcts/SolveXcts.hpp:85]: strange; doesn't show up on your branch when I build it...

**Review by kidder — APPROVED** [review]

**Pull request merged**